File: frontend/src/types/task.ts
// Must match backend enum values in backend/src/models/task.py
export type TaskOutputFormat = 'text' | 'json' | 'pydantic';

export interface Task {
  id: number;
  name: string;
  description: string;
  expected_output: string;
  agent_id?: number;
  crew_id?: number;
  order: number;
  async_execution: boolean;
  output_format: TaskOutputFormat;
  output_file?: string;
  context?: string;
  tools_config?: string;
  created_at: string;
  updated_at: string;
}

export interface TaskCreate {
  name: string;
  description: string;
  expected_output: string;
  agent_id?: number;
  crew_id?: number;
  order?: number;
  async_execution?: boolean;
  output_format?: TaskOutputFormat;
  output_file?: string;
  context?: string;
  tools_config?: string;
}

export interface TaskUpdate {
  name?: string;
  description?: string;
  expected_output?: string;
  agent_id?: number;
  crew_id?: number;
  order?: number;
  async_execution?: boolean;
  output_format?: TaskOutputFormat;
  output_file?: string;
  context?: string;
  tools_config?: string;
}

File: backend/src/schemas/task.py
"""Task schemas for API validation."""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from ..models.task import TaskOutputFormat


class TaskBase(BaseModel):
    """Base task schema."""

    name: str = Field(..., min_length=1, max_length=255)
    description: str = Field(..., min_length=1)
    expected_output: str = Field(..., min_length=1)
    agent_id: Optional[int] = None
    crew_id: Optional[int] = None
    order: int = Field(default=0, ge=0)
    async_execution: bool = False
    output_format: TaskOutputFormat = TaskOutputFormat.TEXT
    output_file: Optional[str] = None
    context: Optional[str] = None
    tools_config: Optional[str] = None
    variables: Optional[List[str]] = None


class TaskCreate(TaskBase):
    """Schema for creating a task."""
    pass


class TaskUpdate(BaseModel):
    """Schema for updating a task."""

    name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = Field(None, min_length=1)
    expected_output: Optional[str] = Field(None, min_length=1)
    agent_id: Optional[int] = None
    crew_id: Optional[int] = None
    order: Optional[int] = Field(None, ge=0)
    async_execution: Optional[bool] = None
    output_format: Optional[TaskOutputFormat] = None
    output_file: Optional[str] = None
    context: Optional[str] = None
    tools_config: Optional[str] = None
    variables: Optional[List[str]] = None


class TaskResponse(TaskBase):
    """Schema for task response."""

    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class TaskListResponse(BaseModel):
    """Schema for paginated task list response."""

    tasks: list[TaskResponse]
    total: int
    page: int = 1
    page_size: int = 10
